src/gramps.js

```diff
@@ -33,23 +33,27 @@ const getDefaultApolloOptions = options => ({
 */
 const mapSourcesToExecutableSchemas = (sources, mock, options) =>
   sources
-    .map(({ schema: typeDefs, resolvers, mocks, namespace }) => {
-      if (!typeDefs) {
+    .map(({ schema, typeDefs, resolvers, mocks, namespace }) => {
+      let executableSchema;
+      if (schema instanceof GraphQLSchema) {
+        executableSchema = schema;
+      } else if (typeof schema === 'string' || typeof typeDefs == 'string'){
+        executableSchema = makeExecutableSchema({
+          typeDefs: typeDefs || schema,
```

Comment:
I think my primary discomfort with this all balls up here. This makes it hard to write docs: "you can provide a `schema` defenition as described in [the GraphQL docs](http://graphql.org/learn/schema/) (which you can also call `typeDefs`), or you can provide a `schema` as generated by [`makeExecutableSchema`](https://www.apollographql.com/docs/graphql-tools/generate-schema.html#makeExecutableSchema), which requires that you supply the GraphQL schema and resolvers to `makeExecutableSchema` inside your data source — but don't worry, we'll do that for you."

That's hard to follow, and someone without a reasonably strong contextual knowledge of both GraphQL's _and_ Apollo's semantics will be completely lost reading these docs. We're effectively asking someone to read two other sets of docs before they can read our docs (or we're duplicating two sets of docs to keep them on-page, which is a bad idea).

The original intention was to limit the number of abstractions someone needed to learn before creating a simple data source. And while we can do a simple quickstart to get around that, we'll end up with the same problem the existing Apollo server docs have, which is that they have a lot of edge case functions that aren't documented with examples, and effectively just let you know they exist and then require that you look at the source and/or just try things to see what the errors look like.

Reply:
My desired API would allow for **EITHER** (`typeDefs` + `resolvers`) **OR** (in *advanced cases*) `schema`

The main reason I even bring this feature up before a 1.0 is because it draws a distinction between `typeDefs` and `schema`

If a 1.0 release used `typeDefs` instead of `schema` I would be happy enough because it's **trivially easy** to allow for data sources to provide an executable `schema` for advanced cases (like remote schemas)

Reply:
Alright — let's do it. Swap out `schema` for `typeDefs`, and let's drop support for executable schemas. I still think it's confusing that we talk about GraphQL schemas, but then refer to them as `typeDefs` in code, but I'll survive.

If we bump up against a use case that requires exposing an executable schema, we can revisit this, but I'd prefer to avoid `OR` options in our config unless there's a _really_ good reason to do it.